### PR TITLE
#1945 - Create shared environment file for ECS tasks

### PR DESCRIPTION
**Pull request by EvanParish**
# Description

This PR removes the environment variables from our task definitions and moves them to environment specific files. Each environment file is uploaded to the designated s3 bucket during deployment.

issue #1945

## How Has This Been Tested?

- [x] verified the env files are uploaded to S3 when deploying the app
- [x] [deployed to dev and the regression passed](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/11387047561) 

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an [appropriate title](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/team_agreement.md#naming-prs): #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/pull_request_labels.md#vanotify-labels) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the [style guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/style_guide.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works. [Testing guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/testing_guide.md)
- [x] I have ensured the latest main is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] The ticket was moved into the DEV test column when I began testing this change

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -29,6 +29,10 @@ jobs:
           role-skip-session-tagging: true
           role-duration-seconds: 900
 
+      - name: Upload Env File to S3
+        run: |
+          aws s3 cp cd/application-deployment/${{ inputs.environment }}/${{ inputs.environment }}.env s3://vanotify-environment-variables-${{ inputs.environment }}/
```
Comment (coreycarvalho) [inline .github/workflows/build.yml:34]: Can we make these filenames unique? I'm imagining a scenario where multiple people are trying to deploy to dev and they overwrite each other's files. When I did this in the past, we used the commit hash as the object name.

File: cd/application-deployment/dev/dev.env [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+TWILIO_AUTH_TOKEN=fake
+UI_HOST_NAME=https://dev.notifications.va.gov
+VANOTIFY_SSL_CERT_PATH=/app/certs/vanotify_ssl_cert.pem
+VANOTIFY_SSL_KEY_PATH=/app/certs/vanotify_ssl_key.pem
+VA_FLAGSHIP_APP_SID=A20623E2321D4053A6C34C9307C6C221
+VA_ONSITE_URL=https://staging-api.va.gov
+VA_PROFILE_EMAIL_STATUS_ENABLED=True
+VA_PROFILE_URL=https://int.vaprofile.va.gov
+VA_PROFILE_V3_COMBINE_CONTACT_INFO_AND_PERMISSIONS_LOOKUP=True
+VA_PROFILE_V3_IDENTIFY_MOBILE_TELEPHONE_NUMBERS=True
```
Comment (cris-oddball) [inline cd/application-deployment/dev/dev.env:52]: we just removed these feature flags yesterday, please remove

File: cd/application-deployment/dev/vaec-celery-task-definition.json [hunk 3]
```diff
@@ -16,218 +16,20 @@
...
-        },
-        {
-          "name": "EMAIL_PASSWORD_LOGIN_ENABLED",
-          "value": "True"
-        },
-        {
-          "name": "EMAIL_PROVIDER_SELECTION_STRATEGY_LABEL",
-          "value": "LOAD_BALANCING"
-        },
         {
```
Comment (coreycarvalho) [inline cd/application-deployment/dev/vaec-celery-task-definition.json:30]: We are keeping this value in the task definition?

File: cd/application-deployment/perf/perf.env [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+UI_HOST_NAME=https://perf.notifications.va.gov
+V3_ENABLED=True
+VANOTIFY_SSL_CERT_PATH=/app/certs/vanotify_ssl_cert.pem
+VANOTIFY_SSL_KEY_PATH=/app/certs/vanotify_ssl_key.pem
+VA_FLAGSHIP_APP_SID=A20623E2321D4053A6C34C9307C6C221
+VA_ONSITE_URL=https://staging-api.va.gov
+VA_PROFILE_EMAIL_STATUS_ENABLED=True
+VA_PROFILE_URL=https://qa.vaprofile.va.gov
+VA_PROFILE_V3_COMBINE_CONTACT_INFO_AND_PERMISSIONS_LOOKUP=True
+VA_PROFILE_V3_IDENTIFY_MOBILE_TELEPHONE_NUMBERS=True
```
Comment (cris-oddball) [inline cd/application-deployment/perf/perf.env:48]: we just removed these feature flags yesterday, please remove

File: cd/application-deployment/prod/prod.env [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+UI_HOST_NAME=https://notifications.va.gov
+V3_ENABLED=False
+VANOTIFY_SSL_CERT_PATH=/app/certs/vanotify_ssl_cert.pem
+VANOTIFY_SSL_KEY_PATH=/app/certs/vanotify_ssl_key.pem
+VA_FLAGSHIP_APP_SID=D288162B716A4F55B9F5CF9AA9DDEA5E
+VA_ONSITE_URL=https://api.va.gov
+VA_PROFILE_EMAIL_STATUS_ENABLED=True
+VA_PROFILE_URL=https://www.vaprofile.va.gov
+VA_PROFILE_V3_COMBINE_CONTACT_INFO_AND_PERMISSIONS_LOOKUP=True
+VA_PROFILE_V3_IDENTIFY_MOBILE_TELEPHONE_NUMBERS=True
```
Comment (cris-oddball) [inline cd/application-deployment/prod/prod.env:48]: we just removed these feature flags yesterday, please remove

File: cd/application-deployment/staging/staging.env [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+UI_HOST_NAME=https://staging.notifications.va.gov
+V3_ENABLED=False
+VANOTIFY_SSL_CERT_PATH=/app/certs/vanotify_ssl_cert.pem
+VANOTIFY_SSL_KEY_PATH=/app/certs/vanotify_ssl_key.pem
+VA_FLAGSHIP_APP_SID=A20623E2321D4053A6C34C9307C6C221
+VA_ONSITE_URL=https://staging-api.va.gov
+VA_PROFILE_EMAIL_STATUS_ENABLED=True
+VA_PROFILE_URL=https://qa.vaprofile.va.gov
+VA_PROFILE_V3_COMBINE_CONTACT_INFO_AND_PERMISSIONS_LOOKUP=True
+VA_PROFILE_V3_IDENTIFY_MOBILE_TELEPHONE_NUMBERS=True
```
Comment (cris-oddball) [inline cd/application-deployment/staging/staging.env:52]: we just removed these feature flags yesterday, please remove

**Review by cris-oddball — CHANGES_REQUESTED** [review]

File: cd/application-deployment/dev/dev.env (hunk 2 above)
Comment (EvanParish) [reply cd/application-deployment/dev/dev.env:52]: Good catch! I'll remove those real quick

File: .github/workflows/build.yml (hunk 1 above)
Comment (coreycarvalho) [reply .github/workflows/build.yml:34]: Spoke to @EvanParish and we decided that naming these things with a sha would actually add more work and complexity for almost no additional value. Resolving this comment.

File: cd/application-deployment/dev/vaec-celery-task-definition.json (hunk 3 above)
Comment (EvanParish) [reply cd/application-deployment/dev/vaec-celery-task-definition.json:30]: There is a separate ticket about removing old or unused variables, this is just moving what's there to the environment files.

**Review by coreycarvalho — APPROVED** [review]

**Review by kalbfled — APPROVED** [review]

**Review by cris-oddball — APPROVED** [review]

**Pull request merged**